### DISCUSSION TITLE
Overwriting animations when reimporting

### DISCUSSION THREAD

**Discussion by FireyJoss** [viniciusgerevini/godot-aseprite-wizard#180] [Q&A]
Hey I am new to godot-aseprite-wizard. I imported some animations with the wizard which works fine, then I changed something small in the animation like e.g. setting the animation to not loop. When I now change something in aseprite and import the animation again all the changes I made in godot are overwritten. I understand why this is the case however it would be practical if the changes made in godot could be taken over when reimporting. Is there a way to do this?

**Comment by viniciusgerevini** [comment]
Hello @FireyJoss. Yeah, this behaviour is by design. The reason for this is that the Aseprite file should be the source of truth. Things can get quite messy when keeping half of things set in the source file and half edited directly in the resource.

I believe as a matter of good practice you should aim to have the Aseprite file as the source of truth and not rely on manual changes. If you need any manual changes that cannot be set via the aseprite file, let me know that it might be a missing feature.

Having said that, if you wish to implement this in your local code, there are a few things that need to be changed:

- First, you need to find a way to detect manual changes. You'd probably have to use the node's / resource's metadata for this, and keep track of the previously imported settings. Then in a new import you should be able to detect which things changed manually.
- Currently, a new resource is created on every import. You'd have to change this to only create a resource when it's empty, and on subsequent imports just update it.
- I believe you will also have to implement a way to "clear" manual changes, otherwise once set, they will remain manual forever. Maybe not a big of a deal for the Loop example, but definitely a headache for things like FPS.

Let me know if this answer your question. Cheers
